signer.js

```diff
@@ -0,0 +1,34 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var jwtool = require('fxa-jwtool')
+
+module.exports = function (secretKeyFile, domain) {
+
+  var key = jwtool.JWK.fromFile(secretKeyFile, {iss: domain })
+
+  return {
+    sign: function (data) {
+      var now = Date.now()
+      return key.sign(
+        {
+          "public-key": data.publicKey,
+          principal: {
+            email: data.email
+          },
+          iat: now - (10 * 1000),
+          exp: now + data.duration,
+          "fxa-generation": data.generation,
+          "fxa-lastAuthAt": data.lastAuthAt,
+          "fxa-verifiedEmail": data.verifiedEmail
```

Comment:
You've dropped some helpful comments on these data fields when copying them across.  Not sure if accidental or deliberate :-)